sklearn/utils/tests/test_validation.py

```diff
@@ -712,7 +725,10 @@ def test_check_dataframe_warns_on_dtype():
                          check_array, df, dtype=np.float64, warn_on_dtype=True)
     assert_warns(DataConversionWarning, check_array, df,
                  dtype='numeric', warn_on_dtype=True)
-    assert_no_warnings(check_array, df, dtype='object', warn_on_dtype=True)
+    with pytest.warns(DeprecationWarning,
```

Comment:
use `pytest.warn(None)` and put this in a separate test if you need to ignore the deprecation warning

Reply:
done